vocabularies/UI.xml

```diff
@@ -1554,15 +1554,22 @@ It is NOT tied to a data value (in contrast to [DataFieldWithIntentBasedNavigati
 
       <ComplexType Name="DataField" BaseType="UI.DataFieldAbstract">
         <Annotation Term="Core.Description" String="A piece of data" />
-        <Property Name="Value" Type="Edm.PrimitiveType" Nullable="false">
+        <Property Name="Value" Type="Edm.Untyped" Nullable="false">
           <Annotation Term="Core.Description" String="The data field's value" />
+          <Annotation Term="Validation.DerivedTypeConstraint">
+            <Collection>
+              <String>Edm.PrimitiveType</String>
+              <String>Collection(Edm.PrimitiveType)</String>
+            </Collection>
+          </Annotation>
           <Annotation Term="Core.IsLanguageDependent" />
         </Property>
       </ComplexType>
 
       <ComplexType Name="DataFieldWithAction" BaseType="UI.DataField">
         <Annotation Term="Core.Description" String="A piece of data that allows triggering an OData action" />
         <Annotation Term="Core.LongDescription" String="The action is tied to a data value which should be rendered as a hyperlink. This is in contrast to [DataFieldForAction](#DataFieldForAction)) which is not tied to a specific data value." />
+        <Property Name="Value" Type="Edm.PrimitiveType" Nullable="false" />
```

Comment:
The specification says that the derived type "inherits" the properties. Does that imply the possiblity that their type is restricted?

Reply:
Correct, see [CSDL, 7 Structural Property](https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_StructuralProperty)

> If a structural property with the same name is defined in any of this type’s base types, then the property’s type MUST be a type derived from the type specified for the property of the base type and constrains this property to be of the specified subtype for instances of this structured type.

`Edm.PrimitiveType` is a subtype of `Edm.Untyped`, so this is allowed.